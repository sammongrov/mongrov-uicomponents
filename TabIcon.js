import React, { Component } from 'react';
import { View, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { Icon } from 'react-native-elements';
import { Colors } from '@ui/theme_default';

const styles = StyleSheet.create({
  container: {
    height: 30,
    width: 30,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default class TabIcon extends Component {
  static propTypes = {
    iconName: PropTypes.string.isRequired,
  };

  static defaultProps = {};

  state = {};

  render() {
    const { iconName } = this.props;
    return (
      <View style={styles.container}>
        <Icon name={iconName} type="material-community" size={22} color={Colors.TEXT_TAB} />
      </View>
    );
  }
}
