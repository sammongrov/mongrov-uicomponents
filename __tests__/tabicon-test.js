import React from 'react';
import renderer from 'react-test-renderer';
// import { Alert } from 'react-native';
import TabIcon from '../TabIcon';

it('TabIcon renders correctly', () => {
  const tree = renderer.create(<TabIcon iconName="wallet" />).toJSON();
  expect(tree).toMatchSnapshot();
});
