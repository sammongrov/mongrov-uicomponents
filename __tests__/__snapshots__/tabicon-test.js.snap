// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabIcon renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 30,
      "justifyContent": "center",
      "width": 30,
    }
  }
>
  <View>
    <View
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#8E8E93",
              "fontSize": 22,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï–„
      </Text>
    </View>
  </View>
</View>
`;
